src/StackExchange.Redis/Configuration/DefaultOptionsProvider.cs

```diff
@@ -0,0 +1,251 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace StackExchange.Redis.Configuration
+{
+    /// <summary>
+    /// A defaults providers for <see cref="ConfigurationOptions"/>.
+    /// This providers defaults not explicitly specified and is present to be inherited by environments that want to provide
+    /// better defaults for their use case, e.g. in a single wrapper library used many places.
+    /// </summary>
+    /// <remarks>
+    /// Why not just have a default <see cref="ConfigurationOptions"/> instance? Good question!
+    /// Since we null coalesce down to the defaults, there's an inherent pit-of-failure with that approach of <see cref="StackOverflowException"/>.
+    /// If you forget anything or if someone creates a provider nulling these out...kaboom.
+    /// </remarks>
+    public class DefaultOptionsProvider
+    {
+        private static readonly List<DefaultOptionsProvider> BuiltInProviders = new()
+        {
+            new AzureOptionsProvider()
+        };
```

Comment:
the one and only time this field is used is: in the KnownInitializers static field initializer; maybe just move this directly inline there?

Reply:
I had it this way before but it's a new nasty/confusing with the shortened `new` syntax it looked harder to understand, IMO - lemme add some comments!